viewing_party/party.py

```diff
@@ -0,0 +1,162 @@
+    return list_movies
+
+    
+def get_available_recs(user_data): 
+    user_watched_info = []
+    index = 0
+    for item in user_data["watched"]:
+        item["host"] = user_data["subscriptions"][index]
+        index +=1
+        user_watched_info.append(item) # list of dict, added host key and value
+    movies_with_sub = []
+    for item in user_data["subscriptions"]:
+        for friend in user_data["friends"]:
+            for subs in friend["watched"]:
+                if item == subs["host"]:
+                    movies_with_sub.append(subs) # list of dict of movies user can watch with their host
+
+    all_friend_movies = []
+    for friends in user_data["friends"]: 
+        for item in friends["watched"]:
+            if item not in all_friend_movies:
+                all_friend_movies.append(item)
+
+    same_movies = []
+    for movie in movies_with_sub: 
+        for friends in all_friend_movies: 
+            if movie["host"] == friends["host"]:
+                if friends not in same_movies:
+                    same_movies.append(friends) # list of possible suggestions based on user host
+
+    suggested_movies = []
+    for movie in same_movies:
+        if movie not in user_watched_info:
+            suggested_movies.append(movie) # final suggestions from friends
+    return suggested_movies
+
+def get_new_rec_by_genre(user_data): 
+    fave = 0
+    fave_genre = None # value of user's favorite genre
+    user_fave_list = []
+    user_fave_count = {}
+    for movie in user_data["watched"]:
+        user_fave_list.append(movie["genre"])
+    for genre in user_fave_list:
+        user_fave_count[genre] = user_fave_list.count(genre)
+        for key, value in user_fave_count.items():
+            if fave < value:
+                fave = value
+                fave_genre = key
+    friend_rec = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["genre"] == fave_genre:
+                if movie not in friend_rec:
+                    friend_rec.append(movie)
+    return friend_rec
+
+def get_rec_from_favorites(user_data):
+    pass
+    user_fave = [] # single variable to loop through user favorites
+    for movie in user_data["favorites"]:
+        user_fave.append(movie)
+
+    fave_seen_by_friends = [] # list of favorites the user and friends have in common
+    for movie in user_data["favorites"]:
+        for item in user_data["friends"]:
+            for friend_movie in item["watched"]:
+                if friend_movie["title"] == movie["title"]:
+                    if movie not in fave_seen_by_friends:
+                        fave_seen_by_friends.append(movie)
+                        
+    suggestion = [] # favorite not seen by any friends, but seen by user
+    for movie in user_fave:
+        if movie not in fave_seen_by_friends:
+            suggestion.append(movie)
+    return suggestion
```

Comment:
💡 Quick-tip: To make your project neater and follow pep8 guidelines, you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.